core/src/main/java/br/com/brainweb/interview/core/features/hero/HeroController.java

```diff
@@ -29,4 +32,60 @@
         final UUID id = heroService.create(createHeroRequest);
         return created(URI.create(format("/api/v1/heroes/%s", id))).build();
     }
+
+    @GetMapping("/id/{id}")
+    public ResponseEntity<?> findById(@PathVariable String id) {
+        try{
+            HeroDTO hero = heroService.findById(UUID.fromString(id));
+            return status(HttpStatus.OK).body(hero);
+        } catch (EmptyResultDataAccessException e) {
+            return status(HttpStatus.NOT_FOUND).body("No Hero found for the given id");
+        }
+    }
+
+    @GetMapping("name/{name}")
+    public ResponseEntity<?> findByName(@Validated @PathVariable String name) {
+        try {
+            return status(HttpStatus.OK).body(heroService.findByName(name));
+        } catch (EmptyResultDataAccessException e) {
+            return status(HttpStatus.OK).body("");
+        }
+    }
+
+    @PutMapping("/{id}")
+    public ResponseEntity<?> updateHero(
+        @Validated @RequestBody HeroDTO hero, @Validated @PathVariable String id
+    ) {
+        try {
+            hero.setId(UUID.fromString(id));
+            heroService.updateHero(hero);
+            return status(HttpStatus.NO_CONTENT).body("");
+        } catch (EmptyResultDataAccessException e) {
+            return status(HttpStatus.NOT_FOUND).body("Cannot update Hero for the given id");
+        }
+
+    }
+
+    @DeleteMapping("/{id}")
+    public ResponseEntity<?> delete(@Validated @PathVariable String id) {
+        try {
+            heroService.delete(UUID.fromString(id));
+            return status(HttpStatus.NO_CONTENT).body("");
+        } catch (EmptyResultDataAccessException e) {
+            return status(HttpStatus.NOT_FOUND).body("Cannot delete Hero for the given id");
+        }
+    }
+
+    @GetMapping("/compare/{id1}/{id2}")
+    public ResponseEntity<?> compare(
+        @Validated @PathVariable("id1") String id1, @Validated @PathVariable("id2") String id2
+    ) {
+        HeroesDTO heroes = heroService.compare(UUID.fromString(id1), UUID.fromString(id2));
+
+        if (heroes == null) {
+            return status(HttpStatus.NO_CONTENT).body("");
+        }
```

Comment:
Não entendi essa lógica.
Se o DTO for null na teoria os ids não existem, correto?
Se não existirem o mais correto é um notFound().build();

204 (No Content) é pra quando a operação da sucesso porém no body não tem retorno, o que não se aplica para esse caso.

Reply:
Sim. É isso mesmo. Acho que acabei confundindo. A intenção era not found mesmo. Não faz sentido retornar no content. hehe.